arrow-core-data/src/test/kotlin/arrow/core/EitherTest.kt

```diff
@@ -203,5 +211,85 @@ class EitherTest : UnitSpec() {
           Right(a).handleErrorWith { Right(b) } == Right(a)
       }
     }
+
+    "suspended Either.fx can bind immediate values" {
+      Gen.either(Gen.string(), Gen.int())
+        .random()
+        .take(1001)
+        .forEach { either ->
+          Either.fx<String, Int> {
+            val res = !either
+            res
+          } shouldBe either
+        }
+    }
+
+    "suspended Either.fx can bind suspended values" {
+      Gen.either(Gen.string(), Gen.int())
+        .random()
+        .take(1001)
+        .forEach { either ->
+          Either.fx<String, Int> {
+            val res = !(suspend {
+              sleep(1)
+              either
+            }).invoke()
+
+            res
+          } shouldBe either
+        }
+    }
+
+    "suspended Either.fx can safely handle immediate exceptions" {
+      Gen.bind(Gen.int(), Gen.throwable(), ::Pair)
+        .random()
+        .take(1001)
+        .forEach { (i, exception) ->
+          shouldThrow<Throwable> {
+            Either.fx<String, Int> {
+              val res = !Either.Right(i)
+              throw exception
+              res
+            }
+            fail("It should never reach here. Either.fx should've thrown $exception")
+          } shouldBe exception
+        }
+    }
+
+    "suspended Either.fx can bind suspended exceptions" {
+      Gen.bind(Gen.int(), Gen.throwable(), ::Pair)
+        .random()
+        .take(1001)
+        .forEach { (i, exception) ->
+          shouldThrow<Throwable> {
+            Either.fx<String, Int> {
+              val res = !Either.Right(i)
+              sleep(1)
```

Comment:
1ms may not be enough to trigger something. It'll make tests slower. Better to have fewer reliable tests.

Reply:
I used 1 `milliseconds` since that's what I've seen work well :D
`ScheduledThreadPoolExecutor` ensures it gets scheduled to resume in `1000` nanos, no that should make `startCoroutineUninterceptedOrReturn` return `COROUTINE_SUSPENDED`.

I'm not sure if there is a better way to ensure suspension here.

So would you be in favor of reducing iterations to `10` with a `100ms` sleep or `100` with an `20ms` sleep instead?